services/core/Ambient/ambient/agent.py

```diff
@@ -0,0 +1,442 @@
+    try:
+        config = utils.load_config(config_path)
+    except Exception:
+        config = {}
+    if not config:
+        _log.error("Ambient agent configuration: ".format(config))
+    if "api_key" not in config:
+        raise RuntimeError("Ambient agent must be configured with an api key.")
+    if "application_key" not in config:
+        raise RuntimeError("Ambient agent must be configured with an "
+                           "application key.")
+    _log.debug("config_dict before init: {}".format(config))
+    utils.update_kwargs_with_config(kwargs, config)
+
+    return Ambient(**kwargs)
+
+
+class Ambient(BaseWeatherAgent):
+    """
+    The Ambient agent requires having an API key to interact with the remote
+    API. The agent offers a performance_mode configuration option which
+    allows users to limit the amount of data returned by the API.
+
+    ***Powered by Dark Sky***
+    """
+
+    def __init__(self, application_key="", **kwargs):
+        super(Ambient, self).__init__(**kwargs)
+        _log.debug("vip_identity: " + self.core.identity)
+        self.headers = {"Accept": "application/json",
+                        "Accept-Language": "en-US"
+                        }
+        self.remove_service("get_hourly_historical")
+        self.remove_service("get_hourly_forecast")
+
+        self.app_key = application_key
+
+        self.last_service_call_timestamp = None
+
+    @RPC.export
+    def get_version(self):
+        """
+        Provides the current version of the agent.
+        :return: current version number in string format.
+        """
+        return __version__
+
+    def validate_location(self, service_name, location):
+        """
+        Indicates whether the location dictionary provided matches the format
+        required by the remote weather API
+        :param service_name: name of the remote API service
+        :param location: location dictionary to provide in the remote API url
+        :return: True if the location matches the required format else False
+        """
+        return isinstance(location.get("location", None), str)
+
+    def get_update_interval(self, service_name):
+        """
+        Indicates the interval between remote API updates
+        :param service_name: requested service endpoint
+        :return: datetime timedelta representing the time interval
+        """
+        if service_name == "get_current_weather":
+            return datetime.timedelta(seconds=3)
+        else:
+            return None
+
+    def get_api_description(self, service_name):
+        """
+        Provides a human-readable description of the various endpoints provided
+        by the agent
+        :param service_name: requested service endpoint
+        :return: Human-readable description string
+        """
+        if service_name is "get_current_weather":
+            "Provides current weather observations by Ambient weather" \
+            " station location name via RPC (Requires " \
```

Comment:
Provides current weather observations by Ambient weather station for a give location